project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js

```diff
@@ -77,8 +79,15 @@ export default class HeatMap extends React.Component {
   componentWillReceiveProps(nextProps) {
     const { placeid, defaultZoom } = this.state;
     const { targetBbox } = this.props;
-
-    if (hasChanged(this.props, nextProps) && nextProps.selectedplace.placeid && placeid !== nextProps.selectedplace.placeid) {
+    const { sharedLink } = nextProps;
+    if (hasChanged(this.props, nextProps) && sharedLink && !this.state.sharedLinkMapRepositions) {
+      const firstBbox = nextProps.bbox.slice(0,2);
+      const secondBbox = nextProps.bbox.slice(2,4);
```

Comment:
Isn't this just the same as `this.getMapBounds(nextProps.bbox)`?

https://github.com/CatalystCode/project-fortis/blob/20289a8bbc183da514ed281790cfdeffe484b554/project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js#L104-L106

Reply:
ahh you're right, i misread the brackets.

Reply:
@c-w actually no, I don't think they're the same. Here's some code I ran to test it... I had to choose the first item in the index.

```
const firstBbox = nextProps.bbox.slice(0,2);
      const secondBbox = nextProps.bbox.slice(2,4);
      const bboxTogether = [firstBbox, secondBbox];
      const bboxTogetherTwo = [this.getMapBounds(nextProps.bbox)][0];
      console.log(bboxTogether);
      console.log('************************');
      console.log(bboxTogetherTwo);
```

and this is what that looked like:

![queries](https://user-images.githubusercontent.com/228230/36573661-169db7b4-1810-11e8-9025-f1a58a791e52.PNG)

Is there something I'm missing, do you think?

Reply:
According to the docs, `fitBounds` takes a tuple of [latitude/longitude bounds](http://leafletjs.com/reference-1.3.0.html#latlngbounds), i.e. the top left and bottom right corners of the rectangle.

The bbox is `[north, west, south, east]` i.e., `[maxY, minX, minY, maxX]` ([see code](https://github.com/CatalystCode/project-fortis/blob/bddc46a2c50ef6edc038e9cdbeffbb076c5358d8/project-fortis-spark/src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dto/SiteSettings.scala#L38-L42) and [docs](https://github.com/CatalystCode/project-fortis/blob/master/project-fortis-pipeline/docs/admin-settings.md#site-settings)). Latitude is y and longitude is x.

Now let's compare the two approaches:

```js
// yours
[bbox.slice(0,2), [bbox.slice(2,4)]
[[bbox[0], bbox[1]], [bbox[2], bbox[3]]
[[maxY, minX], [minY, maxX]

// getMapBounds
[[bbox[1], bbox[0]], [bbox[3], bbox[2]]]
[[minX, maxY], [maxX, MinY]]
```

One of the two must be wrong and we seem to be pretty inconsistent in how we convert between the bbox and the bounds. E.g. renderRectangle follows your approach https://github.com/CatalystCode/project-fortis/blob/20289a8bbc183da514ed281790cfdeffe484b554/project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js#L129-L138 as does changeMapBoundsWithTile https://github.com/CatalystCode/project-fortis/blob/20289a8bbc183da514ed281790cfdeffe484b554/project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js#L114-L119

Could you try changing the implementation of getMapBounds and see it everything still works fine? We should really make all of these consistent and have a single way to convert between bounds and bbox (e.g. currently formatLeafletBounds and getMapBounds are identical!).